### PR TITLE
Switch to tox from rake

### PR DESCRIPTION
**Pull request by aoskotsky-amplify**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-python:
-- "2.7"
+python: 2.7
 install:
-- pip install -r requirements.pip --upgrade
-- pip install -r test-requirements.pip --upgrade
+  - pip install tox-travis
+  - pip install codacy-coverage
 script:
+- mkdir reports # workaround for travis error when writing to this dir
```
Comment (jconstance-amplify) [inline .travis.yml]: we could just commit the directory?

File: setup.py [hunk 2]
```diff
@@ -62,8 +63,4 @@ def get_requirements():
     install_requires=get_requirements(),
     scripts=['bin/dk_monitor'],
     test_suite='nose.collector',
-    entry_points="""
```
Comment (jconstance-amplify) [inline setup.py:65]: So was this stuff that we needed for rake? Or just cruft?

Comment (aoskotsky-amplify) [reply setup.py:65]: It was cruft

File: .travis.yml (hunk 1 above)
Comment (aoskotsky-amplify) [reply .travis.yml]: Its an empty directory though

Comment (jconstance-amplify) [reply .travis.yml]: But apparently we always need it. :P

**Review by jconstance-amplify — APPROVED** [review]

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    DATA_KENNEL_APP_KEY=mock_app_key
+deps=-rtest-requirements.pip
+commands=nosetests
+
+[testenv:lint]
+basepython=python2.7
+deps=-rtest-requirements.pip
+commands=
+    pylint --rcfile=pylintrc --output-format=colorized data_kennel test
+    pycodestyle --config pep8rc
```
Comment (jconstance-amplify) [inline tox.ini]: you need to limit pycodestyle to just the `data_kennel` and `test` directories.

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[tox]
+envlist=py27, lint
+
+[testenv]
+setenv =
+    PYTHONHASHSEED=0
+    DATADOG_API_KEY=mock_api_key
+    DATA_KENNEL_APP_KEY=mock_app_key
+deps=-rtest-requirements.pip
+commands=nosetests
```
Comment (jconstance-amplify) [inline tox.ini]: same problem here.

File: tox.ini (hunk 3 above)
Comment (jconstance-amplify) [inline tox.ini]: Also weren't you going to move the pep8rc stuff to this file?

Comment (aoskotsky-amplify) [reply tox.ini]: i think it belongs in `setup.cfg` instead of here

**Pull request merged**